src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java

```diff
@@ -285,14 +274,48 @@ private static String diffExpectedQueryParams(Request request, JsonNode scenario
 		private static String diffExpectedQueryParam(Request request, Map.Entry<String, JsonNode> queryParam) {
 			QueryParameter requestParam = request.queryParameter(queryParam.getKey());
 			if(requestParam == null || requestParam.key() == null || requestParam.key().isEmpty()){
-				return String.format("Query Parameters: Expected %s, but not found. ", queryParam.getKey());
+				return String.format("Expected Query Parameters: Expected %s, but not found. ", queryParam.getKey());
 			}
 
 			if (!requestParam.containsValue(queryParam .getValue().asText())) {
-				return formatAssert("Query Parameter - " + queryParam.getKey(), queryParam.getValue().asText(), requestParam.firstValue());
+				return formatAssert("Expected Query Parameter:" + queryParam.getKey(), queryParam.getValue().asText(), requestParam.firstValue());
 			}
 
 			return "";
 		}
+
+
+		private static String diffUnexpectedQueryParams(Request request, JsonNode scenarioQueryParams) {
+
+			StringBuilder queryParamDiff = new StringBuilder();
+
+			String requestQueryParamString = request.getUrl().split("\\?")[1];
```

Comment:
What happens if split returns a list with a single element? Could you maybe check the array length in the if statement below and then extract the query param string if it exists